ext/openssl/ossl_pkey.c

```diff
@@ -683,6 +748,30 @@ ossl_pkey_private_to_pem(int argc, VALUE *argv, VALUE self)
     return do_pkcs8_export(argc, argv, self, 0);
 }
 
+/*
+ *  call-seq:
+ *     key.private_to_raw   => string
+ *
+ *  See the OpenSSL documentation for EVP_PKEY_get_raw_private_key()
+ */
+static VALUE ossl_pkey_private_to_raw(VALUE self)
+{
+    EVP_PKEY *pkey;
+    VALUE str;
+    size_t len;
+
+    GetPKey(self, pkey);
+    len = EVP_PKEY_size(pkey);
```

Comment:
The key size doesn't necessary match EVP_PKEY_size(). For example, HMAC pkey type can have an arbitrary long key size but EVP_PKEY_size() will return a fixed number.

According to the manpage, the size needs to be determined by a EVP_PKEY_get_raw_private_key() call with NULL.